src/io_array.ml

```diff
@@ -79,8 +79,8 @@ module Make (IO : Io.S) (Elt : ELT) :
     (* The prefetched area should not exceed 4096 in most cases, thanks to the
        fan out. However, if the hash function is not well distributed, some
        exceptions might happen where the prefetched area actually exceeds 4096.
-       As long as this excess is reasonable (x2), we still want to prefetch. *)
-    2 * 4096
+       As long as this excess is reasonable (x8), we still want to prefetch. *)
```

Comment:
does it force all reads to read 8 full pages?

Reply:
It should not, the amount it reads depends on what the fan-out says: https://github.com/mirage/index/blob/master/src/io_array.ml#L88-L91, it's just an upper bound.

Reply:
Are we loading this buffer on every `find`?

Reply:
Not sure what you mean by "load", but every `find` that has to look in the `data` file reads the disk to (partially) fill it, then (interpolation-)searches into it indeed.